### ISSUE TITLE
About parallel implementation

### ISSUE THREAD

**Issue by TudouJack** [saranshmanu/Federated-Learning#2]
hello! I feel puzzled about the two functions train() and update()

def update(data, target, model, optimizer):
    model.send(data.location)
    optimizer.zero_grad()
    prediction = model(data)
    loss = F.mse_loss(prediction.view(-1), target)
    loss.backward()
    optimizer.step()
    return model

def train():
    for data_index in range(len(remote_dataset[0])-1):
        for remote_index in range(len(compute_nodes)):
            data, target = remote_dataset[remote_index][data_index]
            models[remote_index] = update(data, target, models[remote_index], optimizers[remote_index])
        for model in models:
            model.get()
        return utils.federated_avg({
            "bob": models[0],
            "alice": models[1]
        })

Can the two virtual workers train parallelly？In the federated learning system, all workers will train their local model parallelly.
I think your code cannot  achieve parallel training. Maybe I undestood it wrong, please tell me how the two function work.
